api/src/main/java/org/openmrs/module/pihmalawi/patient/impl/YendaNafePatientRegistrationServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+ */
+@Component
+public class YendaNafePatientRegistrationServiceImpl implements YendaNafePatientRegistrationService {
+    @Authorized( { PrivilegeConstants.ADD_PATIENTS })
+    @Transactional
+    @Override
+    public Patient createPatient(YendaNafePatientRegistrationModel yendaNafePatientRegistrationModel, String creatorUUID, PersonService personService, PatientService patientService, UserService userService, LocationService locationService,String patientIdentifierUuid) {
+        User user = userService.getUserByUuid(creatorUUID);
+
+        String fullName = yendaNafePatientRegistrationModel.name;
+        String[] nameSplitted = fullName.split("\\s+",2);
+        String firstName = nameSplitted[0];
+        String lastName = nameSplitted[1];
+        String gender;
+        if(yendaNafePatientRegistrationModel.sex.toLowerCase().trim().equals("male"))
+        {
+            gender = "M";
+        }
+        else if(yendaNafePatientRegistrationModel.sex.toLowerCase().trim().equals("female"))
+        {
+            gender = "F";
+        }
+        else
+        {
+            gender = "U";
+        }
+
+        SortedSet<PersonName> personNames = new TreeSet<PersonName>();
+
+        PersonName personName = new PersonName();
+        personName.setGivenName(firstName);
+        personName.setFamilyName(lastName);
+        personNames.add(personName);
+
+        SortedSet<PersonAddress> personAddresses = new TreeSet<PersonAddress>();
+        PersonAddress personAddress = new PersonAddress();
+        personAddress.setCountyDistrict("Malawi");
+        personAddress.setCountyDistrict("Neno");
+        personAddress.setCityVillage(yendaNafePatientRegistrationModel.village);
+        personAddresses.add(personAddress);
+
+        Person newPerson = new Person();
+        newPerson.setGender(gender);
+        newPerson.setNames(personNames);
+        newPerson.setCreator(user);
+        newPerson.setBirthdate(DateUtil.parseYmd(yendaNafePatientRegistrationModel.date_of_birth));
+        newPerson.setAddresses(personAddresses);
+        newPerson.addName(personName);
+
+        Person savedPerson = personService.savePerson(newPerson);
+        Location patient_location = locationService.getLocationByUuid(yendaNafePatientRegistrationModel.location_uuid);
+        PatientIdentifierType patientIdentifierType =  patientService.getPatientIdentifierTypeByUuid(patientIdentifierUuid);
+        PatientIdentifier patientIdentifier = new PatientIdentifier();
+        SortedSet<PatientIdentifier> patientIdentifiers = new TreeSet<PatientIdentifier>();
+        patientIdentifier.setIdentifier(yendaNafePatientRegistrationModel._id);
+        patientIdentifier.setIdentifierType(patientIdentifierType);
+        patientIdentifier.setLocation(patient_location);
+        patientIdentifiers.add(patientIdentifier);
+        Patient newPatient = new Patient(savedPerson);
+        newPatient.setIdentifiers(patientIdentifiers);
```

Comment:
Is there not an "addIdentifier" method you can use instead of creating and setting an entire set?

Reply:
@mseaton Have checked for the method, it is there. Its better than setting the entire set.  Will replace the sortedSets with addIntentifier, addAddress,addNames. Thanks.